cro-clib/src/jsonrpc.rs

```diff
@@ -0,0 +1,181 @@
+};
+
+use crate::types::CroResult;
+
+#[cfg(not(feature = "mock-enc-dec"))]
+type AppTransactionCipher = DefaultTransactionObfuscation;
+#[cfg(feature = "mock-enc-dec")]
+type AppTransactionCipher = MockAbciTransactionObfuscation<WebsocketRpcClient>;
+
+type AppTxBuilder = DefaultWalletTransactionBuilder<SledStorage, LinearFee, AppTransactionCipher>;
+type AppWalletClient = DefaultWalletClient<SledStorage, WebsocketRpcClient, AppTxBuilder>;
+type AppOpsClient = DefaultNetworkOpsClient<
+    AppWalletClient,
+    SledStorage,
+    WebsocketRpcClient,
+    LinearFee,
+    AppTransactionCipher,
+>;
+type AppSyncerConfig =
+    ObfuscationSyncerConfig<SledStorage, WebsocketRpcClient, AppTransactionCipher>;
+
+/// normal
+#[cfg(not(feature = "mock-enc-dec"))]
+fn get_tx_query(tendermint_client: WebsocketRpcClient) -> Result<DefaultTransactionObfuscation> {
+    DefaultTransactionObfuscation::from_tx_query(&tendermint_client)
+}
+
+/// temporary
+#[cfg(feature = "mock-enc-dec")]
+fn get_tx_query(
+    tendermint_client: WebsocketRpcClient,
+) -> Result<MockAbciTransactionObfuscation<WebsocketRpcClient>> {
+    Ok(MockAbciTransactionObfuscation::new(tendermint_client))
+}
+
+fn make_wallet_client(
+    storage: SledStorage,
+    tendermint_client: WebsocketRpcClient,
+) -> Result<AppWalletClient> {
+    let signer_manager = WalletSignerManager::new(storage.clone());
+    let transaction_cipher = get_tx_query(tendermint_client.clone())?;
+    let transaction_builder = DefaultWalletTransactionBuilder::new(
+        signer_manager,
+        tendermint_client.genesis().unwrap().fee_policy(),
+        transaction_cipher,
+    );
+    Ok(DefaultWalletClient::new(
+        storage,
+        tendermint_client,
+        transaction_builder,
+        Some(50),
+    ))
+}
+
+fn make_ops_client(
+    storage: SledStorage,
+    tendermint_client: WebsocketRpcClient,
+) -> Result<AppOpsClient> {
+    let transaction_cipher = get_tx_query(tendermint_client.clone())?;
+    let signer_manager = WalletSignerManager::new(storage.clone());
+    let fee_algorithm = tendermint_client.genesis().unwrap().fee_policy();
+    let wallet_client = make_wallet_client(storage, tendermint_client.clone())?;
+    Ok(DefaultNetworkOpsClient::new(
+        wallet_client,
+        signer_manager,
+        tendermint_client,
+        fee_algorithm,
+        transaction_cipher,
+    ))
+}
+
+fn make_syncer_config(
+    storage: SledStorage,
+    tendermint_client: WebsocketRpcClient,
+) -> Result<AppSyncerConfig> {
+    let transaction_cipher = get_tx_query(tendermint_client.clone())?;
+
+    Ok(AppSyncerConfig::new(
+        storage,
+        tendermint_client,
+        transaction_cipher,
+        true,
+        50,
+        50,
+    ))
+}
+
+fn do_jsonrpc_call(
+    storage_dir: &str,
+    websocket_url: &str,
+    network_id: u8,
+    json_request: &str,
+) -> Result<String> {
+    let mut io = IoHandler::new();
+    let storage = SledStorage::new(storage_dir)?;
```

Comment:
how de-coupling storage?
so that user supply other database for storage

Reply:
That's a potential feature for the future.